### ISSUE TITLE
Any reason why this isn't on crates.io ? 

### ISSUE THREAD

**Issue by midnightexigent** [neovide/neovide#839]
This would allow to install via `cargo install neovide`

**Comment by Kethku** [comment]
Thats a good recommendation. Just haven't done it yet.

**Comment by Kethku** [comment]
Ok I looked into this. We currently use 2 forks under my personal repository:
1. nvim-rs which I forked to serialize events to and from neovim. There has been a bunch of progress here: https://github.com/KillTheMule/nvim-rs/pull/26 which I find promising
2. glutin/winit which I forked to get an early version of the keyboard handling api. This should merge in the not too distant future

Turns out you can't publish a crate until all of the dependencies are also on crates.io so this is blocked until those upstream changes merge

**Comment by Kethku** [comment]
https://github.com/neovide/neovide/pull/867 solves issue 1 but is waiting on an upstream PR to merge which should happen pretty soon. Issue 2 is still blocked but there has been recent progress on the linux side which means it shouldn't be too much longer.

**Comment by Kethku** [comment]
No update. However a bunch of commits just went in to the linux branch so there is hope

**Comment by fredizzimo** [comment]
We also have some, or at least my fixes, that we need to push upstream first. Do you think we can say that the fix has been working well @Kethku? I can try to send a pull request against their windows branch. I still don't remember exactly what happened, but maybe something along the explanation I gave to you, along with a description of what problems it has solved here is good enough for them? Otherwise I have to try to reproduce the bug and debug it again, to give them a better explanation.

**Comment by Kethku** [comment]
I haven't encountered a lock since merging your changes so I'd say its working great :)

As far as an upstream pull request, I honestly don't know what the correct next step is. We've merged PRs against the individual keyboard branches in the past with mixed results. It puts pressure on the folks who opened the PRs rather than the main maintainers.

Maybe the correct move is to make a PR on the main upstream winit branch explaining the situation and then work on a minimal repro if they ask for one rather than doing the work ahead of time. Then once thats merged upstream we can look into what it would take to rebase the keyboard branches on top of that.

I can maybe take a stab at that in the coming weeks if you'd prefer. You've already done a bunch of work :)

**Comment by fredizzimo** [comment]
Well, the bug is in the keyboard branch, The file `src/platform_impl/windows/keyboard.rs` does not even exist on the master. I haven't checked if that's because of some refactoring. But even if it is, it would be still be work for the people that made the pull request, since they would have to resolve the conflicts.

BTW, the just opened a new pull request with the Windows code rebased against master. I think we can ask them there what to do, just linking our pull request, and the related issues. Do you want to do that?

**Comment by Kethku** [comment]
I totally didn't realize it was keyboard branch specific. By all means this is the correct course of action. Carry on :)

**Comment by Kethku** [comment]
I will spend some time tonight pulling the relevant details, if you do not. Either way works. (don't want to imply any needed work from you)

**Comment by fredizzimo** [comment]
I can do it now, but I think I will go to bed after that, so if you can check if they need more info, then we are good.

**Comment by Kethku** [comment]
Will do. I'll look for your PR tonight and respond if needed

**Comment by MultisampledNight** [comment]
(sidenote since I've seen the just referring issue: you can still install neovide through `cargo install --git https://github.com/neovide/neovide` almost as easy)

**Comment by fredizzimo** [comment]
It is now!
